relayer/spy_relayer/src/relayer/solana_utils.ts

```diff
@@ -0,0 +1,597 @@
+    programId,
+  );
+};
+
+const getOwnerTokenAccountsForPool = async (
+  pool: PublicKey,
+  owner: PublicKey,
+  twoPoolProgram: Program<TwoPool>,
+): Promise<readonly PublicKey[]> => {
+  const tokenIdMapPoolData = await twoPoolProgram.account.twoPool.fetch(pool);
+  const tokenIdMapPoolInfo = {
+    pool,
+    tokenMints: tokenIdMapPoolData.tokenMintKeys,
+    tokenAccounts: tokenIdMapPoolData.tokenKeys,
+    lpMint: tokenIdMapPoolData.lpMintKey,
+    governanceFeeAcct: tokenIdMapPoolData.governanceFeeKey,
+  };
+  const mints = [...tokenIdMapPoolInfo.tokenMints, tokenIdMapPoolInfo.lpMint];
+  return await Promise.all(
+    mints.map(async (mint) => {
+      return await Token.getAssociatedTokenAddress(ASSOCIATED_TOKEN_PROGRAM_ID, TOKEN_PROGRAM_ID, mint, owner);
+    }),
+  );
+};
+
+type WormholeAddresses = {
+  readonly wormhole: PublicKey;
+  readonly tokenBridge: PublicKey;
+  readonly custody: PublicKey;
+  readonly wormholeConfig: PublicKey;
+  readonly wormholeFeeCollector: PublicKey;
+  readonly wormholeEmitter: PublicKey;
+  readonly wormholeSequence: PublicKey;
+  readonly authoritySigner: PublicKey;
+  readonly tokenBridgeConfig: PublicKey;
+  readonly custodySigner: PublicKey;
+};
+
+type MarginalPricePoolInfo = {
+  readonly pool: PublicKey;
+  readonly token0Account: PublicKey;
+  readonly token1Account: PublicKey;
+  readonly lpMint: PublicKey;
+};
+
+export const getWormholeAddressesForMint = async (
+  wormhole: web3.PublicKey,
+  tokenBridge: web3.PublicKey,
+  mint: web3.PublicKey,
+): Promise<WormholeAddresses> => {
+  const [custody] = await (async () => {
+    return await web3.PublicKey.findProgramAddress(
+      [mint.toBytes()],
+      tokenBridge,
+    );
+  })();
+
+  const [wormholeConfig] = await web3.PublicKey.findProgramAddress(
+    [Buffer.from("Bridge")],
+    wormhole,
+  );
+  const [wormholeFeeCollector] = await web3.PublicKey.findProgramAddress(
+    [Buffer.from("fee_collector")],
+    wormhole,
+  );
+
+  const [wormholeEmitter] = await web3.PublicKey.findProgramAddress(
+    [Buffer.from("emitter")],
+    tokenBridge,
+  );
+  const [wormholeSequence] = await web3.PublicKey.findProgramAddress(
+    [Buffer.from("Sequence"), wormholeEmitter.toBytes()],
+    wormhole,
+  );
+
+  const [authoritySigner] = await web3.PublicKey.findProgramAddress(
+    [Buffer.from("authority_signer")],
+    tokenBridge,
+  );
+  const [tokenBridgeConfig] = await web3.PublicKey.findProgramAddress(
+    [Buffer.from("config")],
+    tokenBridge,
+  );
+  const [custodySigner] = await web3.PublicKey.findProgramAddress(
+    [Buffer.from("custody_signer")],
+    tokenBridge,
+  );
+  return {
+    wormhole,
+    tokenBridge,
+    custody,
+    wormholeConfig,
+    wormholeFeeCollector,
+    wormholeEmitter,
+    wormholeSequence,
+    authoritySigner,
+    tokenBridgeConfig,
+    custodySigner,
+  };
+};
+
+export const generatePropellerEngineTxns = async (
```

Comment:
This naming sounds like you're using a generator. Suggest:
```suggestion
export const createPropellerEngineTxns = async (
```